source/Server/Configuration/JiraConfigurationSettings.cs

```diff
@@ -31,9 +31,9 @@ public override IEnumerable<IConfigurationValue> GetConfigurationValues()
 
             yield return new ConfigurationValue<bool>(  "Octopus.JiraIntegration.IsEnabled", isEnabled, isEnabled, "Is Enabled");
             yield return new ConfigurationValue<string>("Octopus.JiraIntegration.BaseUrl", ConfigurationDocumentStore.GetBaseUrl(), isEnabled && !string.IsNullOrWhiteSpace(ConfigurationDocumentStore.GetBaseUrl()), "Jira Base Url");
-            yield return new ConfigurationValue<string>("Octopus.JiraIntegration.ConnectAppPassword", ConfigurationDocumentStore.GetConnectAppPassword(), isEnabled && !string.IsNullOrWhiteSpace(ConfigurationDocumentStore.GetConnectAppPassword()), "Jira Connect App Password", isSensitive: true);
+            yield return new ConfigurationValue<string>("Octopus.JiraIntegration.ConnectAppPassword", ConfigurationDocumentStore.GetConnectAppPassword()?.Value, isEnabled && !string.IsNullOrWhiteSpace(ConfigurationDocumentStore.GetConnectAppPassword()?.Value), "Jira Connect App Password", isSensitive: true);
```

Comment:
As discussed, it would be nice to have another constructor, or a "static constructor" to handle sensitive values.